system-contract-dapp-playground/src/components/contract-interaction/hts/token-create-custom/methods/AssociateHederaToken.tsx

```diff
@@ -0,0 +1,209 @@
+/*-
+ *
+ * Hedera Smart Contracts
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import Cookies from 'js-cookie';
+import { Contract } from 'ethers';
+import { useToast } from '@chakra-ui/react';
+import { useState, useEffect } from 'react';
+import { CommonErrorToast } from '@/components/toast/CommonToast';
+import { TransactionResult } from '@/types/contract-interactions/HTS';
+import { handleAPIErrors } from '../../shared/methods/handleAPIErrors';
+import { TRANSACTION_PAGE_SIZE } from '../../shared/states/commonStates';
+import { useToastSuccessful } from '../../shared/hooks/useToastSuccessful';
+import { usePaginatedTxResults } from '../../shared/hooks/usePaginatedTxResults';
+import { TransactionResultTable } from '../../shared/components/TransactionResultTable';
+import { associateHederaTokensToAccounts } from '@/api/hedera/tokenCreateCustom-interactions';
+import { handleSanitizeHederaFormInputs } from '../../shared/methods/handleSanitizeFormInputs';
+import { SharedFormInputField, SharedExecuteButton } from '../../shared/components/ParamInputForm';
+import { useUpdateTransactionResultsToLocalStorage } from '../../shared/hooks/useUpdateLocalStorage';
+import { htsTokenAssociateParamFields } from '@/utils/contract-interactions/HTS/token-create-custom/constant';
+import { handleRetrievingTransactionResultsFromLocalStorage } from '../../shared/methods/handleRetrievingTransactionResultsFromLocalStorage';
+
+interface PageProps {
+  baseContract: Contract;
+}
+
+const AssociateHederaToken = ({ baseContract }: PageProps) => {
```

Comment:
Instead of comma separate values in one field we should make it an expandable list list you did for keys.
That is have one field for a token address and with every `+` button click you add another UI element for a token field